logistics_planning_invoicing/models/logistics_schedule.py

```diff
@@ -0,0 +1,86 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class LogisticsSchedule(models.Model):
+    _inherit = 'logistics.schedule'
+
+    account_move_line_id = fields.Many2one('account.move.line', readonly=True)
+    account_move_id = fields.Many2one('account.move', related='account_move_line_id.move_id', string='Invoice')
+    is_invoiceable = fields.Boolean(compute='_compute_is_invoiceable', store=True, readonly=False)
+
+
+    @api.depends('account_move_line_id', 'state')
+    def _compute_is_invoiceable(self):
+        for record in self:
+            if record.account_move_line_id or record.state == 'done':
+                record.is_invoiceable = False
+            else:
+                record.is_invoiceable = True
+
+    # def is_invoiceable(self):
+    #     return not any(
+    #         record.account_move_line_id or record.state == 'done'
+    #         for record in self
+    #     )
+
+    def action_mark_done(self):
+        self.state = 'done'
+
+    def action_create_invoice_wizard(self):
+        if not self.env.user.has_group("account.group_account_user") and not self.env.user.has_group("account.group_account_manager"):
+            raise ValidationError(_('You need invoice user access to perform this action.'))
+
+        lg_ids = self.browse(self.env.context.get('active_ids')).filtered(lambda x: x.is_invoiceable)
+        carrier_id = lg_ids.mapped('carrier_id')
+
+        if not lg_ids:
+            raise ValidationError(_('You cannot select records with a done status or with an invoice already assigned.'))
```

Comment:
This error message is confusing. Try this:
```suggestion
            raise ValidationError(_('No invoiceable schedules were selected'))
```